src/scheduler/batch/main.ts

```diff
@@ -1,95 +1,29 @@
+import { updateListsForAnnualReview } from "./tasks/updateListsForAnnualReview";
 
-export async function populateCurrentAnnualReview(lists: List[]): Promise<void> {
-  const listIds = lists.map((list) => list.id);
-  const findListItemsResult = await findListItems({
-    listIds,
-    statuses: ["PUBLISHED", "CHECK_ANNUAL_REVIEW"],
-    isAnnualReview: false,
-  });
-  if (findListItemsResult.error) {
-    logger.error(`Unable to retrieve List Items for Lists ${listIds}: ${findListItemsResult.error.message}`);
-    return;
-  }
-  if (!findListItemsResult.result.length) {
-    logger.info(`No List Items found for Lists ${listIds}`);
-    return;
-  }
-  const listItemsForAllLists: ListItemWithHistory[] = findListItemsResult.result;
-  for (const list of lists) {
-    // exclude list items published in the last month from annual review
-    const listItemsEligibleForAnnualReview = listItemsForAllLists.filter((listItem) => {
-      if (!listItem.history.length) {
-        return false;
-      }
-      return listItem.listId === list.id;
-    });
-    if (!listItemsEligibleForAnnualReview.length) {
-      logger.info(`No list items identified for list ${list.id}, excluding from sending annual review emails`);
-    } else {
-      const listItemIdsForAnnualReview = listItemsEligibleForAnnualReview.map((listItem) => listItem.id);
+async function main() {
+  // if a task needs to be executed first, await them here.
+  logger.info("starting populateMissingAnnualReviewDates");
+  await populateMissingAnnualReviewDates();
 
-      const contexts = helpers.getDateContexts(list.nextAnnualReviewStartDate);
-      const currentAnnualReview = getCurrentAnnualReviewData(listItemIdsForAnnualReview, contexts);
-      let isUpdateList = true;
-      // use the same reference if it already exists to ensure audit records can be checked for previous emails
-      if (list.jsonData?.currentAnnualReview?.reference) {
-        const { eligibleListItems: currentEligibileListItems } = list.jsonData.currentAnnualReview;
-        const { eligibleListItems: newEligibileListItems } = currentAnnualReview;
-        isUpdateList = !_.isEqual(
-          currentEligibileListItems.sort((a, b) => a - b),
-          newEligibileListItems.sort((a, b) => a - b)
-        );
-        if (isUpdateList) {
-          currentAnnualReview.reference = list.jsonData.currentAnnualReview.reference;
-          currentAnnualReview.keyDates = list.jsonData.currentAnnualReview.keyDates;
-        }
-      }
+  logger.info("starting updateListsForAnnualReview");
+  await updateListsForAnnualReview(startOfToday());
 
-      if (isUpdateList) {
-        await updateListForAnnualReview(list, { currentAnnualReview });
-      }
-    }
-  }
+  // put all the tasks that can be run async (i.e. not required to be sequential) here.
+  /*
+   const tasks = []; // needs to be an array of promises.
+   return await Promise.allSettled(tasks);
+   */
```

Comment:
How necessary is this comment?

Reply:
It's there to match the worker workflow (so there is a standardised way of dealing with things). Whilst not necessary, saves you from searching around and checking if there's a way to run tasks sequentially vs non sequential, and matching how this project does it.

Reply:
fair enough